examples/navier-stokes/nsplex.c

```diff
@@ -1028,10 +1031,34 @@ int main(int argc, char **argv) {
   user->dt = dt;
 
   // Solve
+  start = MPI_Wtime();
   ierr = TSSolve(ts, Q); CHKERRQ(ierr);
+  ierr = TSGetSolveTime(ts,&ftime); CHKERRQ(ierr);
+  cpu_time_used = MPI_Wtime() - start;
+  ierr = PetscPrintf(PETSC_COMM_WORLD,
+                    "Time taken for solution: %g\n",
+                    (double)cpu_time_used); CHKERRQ(ierr);
```

Comment:
You'll want an `MPI_Allreduce` here

Example from `bps.c`:
`
        ierr = MPI_Allreduce(&my_rt, &rt_max, 1, MPI_DOUBLE, MPI_MAX, comm);
        CHKERRQ(ierr);`